### PR TITLE
Response/Rendering fixes

### PR DESCRIPTION
**Pull request by vangheem**
- `guillotina.browser.Response` moved to `guillotina.response.Response`
- move `guillotina.browser.ErrorResponse` to `guillotina.response.ErrorResponse`
- `guillotina.browser.UnauthorizedResponse` removed
- `guillotina.response.Response` no longer supports wrapping aiohttp responses
- `guillotina.response.Response` can now be raised as an exception
- return or raised aiohttp responses now bypass guillotina renderer framework
- raising any Response as an exception aborts current transaction
- remove `IFrameFormatsJson`
- remove `IRenderFormats`, `IRendered` is now a named adapter lookup
- remove `app_settings.renderers` setting. Use the lookups
- remove `IDownloadView`
- remove `TraversableDownloadService`
- remove `IForbiddenAttribute`
- remove `ISerializableException`
- remove `IForbidden`

### CODE REVIEW

File: guillotina/contentnegotiation.py [hunk 1]
```diff
@@ -1,844 +1,34 @@
...
+    ca-ES,ca;q=0.9,en;q=0.8,es;q=0.7
+    '''
+    result = []
+    for ct in value.lower().split(','):
+        weight = 1.0
+        ct_name = ct.split(';')[0]
+        for param in ct.split(';')[1:]:
+            if param.startswith('q='):
+                try:
+                    weight = float(param[len('q='):])
```
Comment (bloodbare) [inline guillotina/contentnegotiation.py]: Can we replace to 2: ?

File: guillotina/contentnegotiation.py [hunk 2]
```diff
@@ -1,844 +1,34 @@
...
+                except ValueError:
+                    pass
+        result.append((weight, ct_name))
+    return [i[1] for i in sorted(result, reverse=True)]
```
Comment (bloodbare) [inline guillotina/contentnegotiation.py]: maybe we could use the _sort_by_q deleted function so we get the real order?

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,7 +1,21 @@
...
-- Nothing changed yet.
+4.0.0 (unreleased)
+------------------
+
+- `guillotina.browser.Response` moved to `guillotina.response.Response`
+- move `guillotina.browser.ErrorResponse` to `guillotina.response.ErrorResponse`
+- `guillotina.browser.UnauthorizedResponse` removed
+- `guillotina.response.Response` no longer supports wrapping aiohttp responses
+- `guillotina.response.Response` can now be raised as an exception
+- return or raised aiohttp responses now bypass guillotina renderer framework
```
Comment (ableeb) [inline CHANGELOG.rst]: typo on return should be 'returned'

File: guillotina/contentnegotiation.py [hunk 4]
```diff
@@ -1,844 +1,34 @@
...
-        return self.cn.negotiate(**kwargs)
-
-
-def content_type_negotiation(request, resource, view):
+def get_ordered_content_type_value_list(value):
+    '''
+    ca-ES,ca;q=0.9,en;q=0.8,es;q=0.7
+    '''
+    result = []
+    for ct in value.lower().split(','):
```
Comment (bloodbare) [inline guillotina/contentnegotiation.py]: We should strip also

File: guillotina/contentnegotiation.py [hunk 5]
```diff
@@ -1,844 +1,34 @@
...
+        weight = 1.0
+        ct_name = ct.split(';')[0]
+        for param in ct.split(';')[1:]:
```
Comment (bloodbare) [inline guillotina/contentnegotiation.py]: There is the use case were accept can be:     Accept: text/*, text/plain, text/plain;format=flowed, */*
[https://tools.ietf.org/html/rfc7231#section-5.3.2](https://tools.ietf.org/html/rfc7231#section-5.3.2)

File: guillotina/contentnegotiation.py [hunk 6]
```diff
@@ -1,844 +1,34 @@
...
-        # if download view, we want to render raw immediately
-        # or if no or */* accept header provided
-        return IRendererFormatRaw
-
-    np = get_utility(IContentNegotiation, 'content_type')
-    ap = np.negotiate(accept=accept)
-    # We need to check for the accept
-    if str(ap.content_type) in app_settings['renderers']:
-        return app_settings['renderers'][str(ap.content_type)]
+        return get_ordered_content_type_value_list(request.headers['ACCEPT'])
```
Comment (bloodbare) [inline guillotina/contentnegotiation.py]: We are losing the difference between type and subtype. Maybe we need two different parsers (like the code we are removing)

File: guillotina/contentnegotiation.py (hunk 1 above)
Comment (vangheem) [reply guillotina/contentnegotiation.py]: awww, I like this more. Compiler takes care of it and makes `'q='` a constant to 2 anyways.

It's a style thing to be explicit about what we're doing.

File: guillotina/contentnegotiation.py (hunk 2 above)
Comment (vangheem) [reply guillotina/contentnegotiation.py]: I'm not familiar with what you're talking about here. Can you clarify?

Comment (bloodbare) [reply guillotina/contentnegotiation.py]: If u check the function from the old code u will see the logic on the priority to define which is prior to which one

Comment (vangheem) [reply guillotina/contentnegotiation.py]: ah yes, I see. I'll fix

Comment (vangheem) [reply guillotina/contentnegotiation.py]: the old code is so horrible

**Pull request merged**